keyboards/preonic/keymaps/mverteuil/keymap.c

```diff
@@ -0,0 +1,511 @@
+            }
+            break;
+        default:
+            break;
+    }
+}
+
+void td_numpad_funcrow_reset(qk_tap_dance_state_t *state, void *user_data) {
+    switch (qk_tap_state.numpad_funcrow) {
+        case SINGLE_HOLD:
+            layer_off(_NUMPAD);
+            break;
+        case DOUBLE_HOLD:
+            layer_off(_FUNCTIONROW);
+            break;
+        default:
+            break;
+    }
+}
+
+void td_brackets_left_finished(qk_tap_dance_state_t *state, void *user_data) {
+    qk_tap_state.left_brackets = get_tapdance_state(state);
+    switch (qk_tap_state.left_brackets) {
+        case SINGLE_TAP:
+            register_left_paren();
+            break;
+        case DOUBLE_TAP:
+            register_code(KC_LBRC);
+            break;
+        case TRIPLE_TAP:
+            register_left_brace();
+            break;
+        default:
+            break;
+    }
+}
+
+void td_brackets_left_reset(qk_tap_dance_state_t *state, void *user_data) {
+    switch (qk_tap_state.left_brackets) {
+        case SINGLE_TAP:
+            unregister_left_paren();
+            break;
+        case DOUBLE_TAP:
+            unregister_code(KC_LBRC);
+            break;
+        case TRIPLE_TAP:
+            unregister_left_brace();
+            break;
+        default:
+            break;
+    }
+    qk_tap_state.left_brackets = 0;
+}
+
+void td_brackets_right_finished(qk_tap_dance_state_t *state, void *user_data) {
+    qk_tap_state.right_brackets = get_tapdance_state(state);
+    switch (qk_tap_state.right_brackets) {
+        case SINGLE_TAP:
+            register_right_paren();
+            break;
+        case DOUBLE_TAP:
+            register_code(KC_RBRC);
+            break;
+        case TRIPLE_TAP:
+            register_right_brace();
+            break;
+        default:
+            break;
+    }
+}
+
+void td_brackets_right_reset(qk_tap_dance_state_t *state, void *user_data) {
+    switch (qk_tap_state.right_brackets) {
+        case SINGLE_TAP:
+            unregister_right_paren();
+            break;
+        case DOUBLE_TAP:
+            unregister_code(KC_RBRC);
+            break;
+        case TRIPLE_TAP:
+            unregister_right_brace();
+            break;
+        default:
+            break;
+    }
+    qk_tap_state.right_brackets = 0;
+}
+
+bool     muse_mode      = false;
+uint8_t  last_muse_note = 0;
+uint16_t muse_counter   = 0;
+uint8_t  muse_offset    = 70;
+uint16_t muse_tempo     = 50;
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (muse_mode) {
+        if (IS_LAYER_ON(_RAISE)) {
+            if (clockwise) {
+                muse_offset++;
+            } else {
+                muse_offset--;
+            }
+        } else {
+            if (clockwise) {
+                muse_tempo += 1;
+            } else {
+                muse_tempo -= 1;
+            }
+        }
+    } else {
+        if (clockwise) {
+            register_code(KC_PGDN);
+            unregister_code(KC_PGDN);
```

Comment:
This can be replaced with `tap_code(KC_PGDN)`, as it will do both of these (plus an optional delay between the two)